compiler/next/lib/uast/Local.cpp

```diff
@@ -28,33 +28,45 @@ namespace uast {
 bool Local::contentsMatchInner(const ASTNode* other) const {
   const Local* lhs = this;
   const Local* rhs = (const Local*) other;
-  return lhs->expressionContentsMatchInner(rhs);
+
+  if (lhs->condChildNum_ != rhs->condChildNum_)
+    return false;
+
+  if (lhs->blockStyle_ != rhs->blockStyle_)
+    return false;
+
+  if (!lhs->expressionContentsMatchInner(rhs))
+    return false;
```

Comment:
Some method somewhere needs to check that the number of children and each child matches. I'd expect that to be in a parent class, perhaps the top AST class. Has this been implemented?

Reply:
That is what `expressionContentsMatchInner` does, yes.